File: sdks/guides/typescript/integrations/llm-groq.ts
// Third-party integration - requires: pnpm add groq-sdk
// See: /guides/ai-agents

import Groq from 'groq-sdk';

const client = new Groq();

// > Groq usage
export async function complete(messages: Array<{ role: string; content: string }>) {
  const r = await client.chat.completions.create({
    model: 'llama-3.3-70b-versatile',
    messages: messages as Groq.ChatCompletionMessageParam[],
    tool_choice: 'auto',
    tools: [
      {
        type: 'function',
        function: {
          name: 'get_weather',
          description: 'Get weather for a location',
          parameters: {
            type: 'object',
            properties: { location: { type: 'string' } },
            required: ['location'],
          },
        },
      },
    ],
  });
  const msg = r.choices[0]?.message;
  const toolCalls = (msg?.tool_calls ?? []).map((tc) => ({
    name: tc.function?.name ?? '',
    args: JSON.parse(tc.function?.arguments ?? '{}'),
  }));
  return {
    content: msg?.content ?? '',
    toolCalls,
    done: toolCalls.length === 0,
  };
}
// !!

File: sdks/guides/python/integrations/llm_groq.py
# Third-party integration - requires: pip install groq
# See: /guides/ai-agents

import json

from groq import Groq

client = Groq()


# > Groq usage
def complete(messages: list[dict]) -> dict:
    r = client.chat.completions.create(
        model="llama-3.3-70b-versatile",
        messages=messages,
        tool_choice="auto",
        tools=[
            {
                "type": "function",
                "function": {
                    "name": "get_weather",
                    "description": "Get weather for a location",
                    "parameters": {
                        "type": "object",
                        "properties": {"location": {"type": "string"}},
                        "required": ["location"],
                    },
                },
            }
        ],
    )
    msg = r.choices[0].message
    tool_calls = [
        {"name": tc.function.name, "args": json.loads(tc.function.arguments or "{}")}
        for tc in (msg.tool_calls or [])
    ]
    return {"content": msg.content or "", "tool_calls": tool_calls, "done": not tool_calls}
# !!
